types/src/traits/election.rs

```diff
@@ -234,6 +269,83 @@ pub trait Election<TYPES: NodeType>: Clone + Eq + PartialEq + Send + Sync + 'sta
         is_valid_signature && is_valid_vote_token
     }
 
+    /// Accumlate the vote, return the QC if the threshold was reached, if not return the updated accumlator
+    #[allow(clippy::type_complexity)]
+    fn accumulate_qc_vote(
+        &self,
+        encoded_key: &EncodedPublicKey,
+        encoded_signature: &EncodedSignature,
+        leaf_commitment: Commitment<Self::LeafType>,
```

Comment:
An idea to reduce code duplication:

Maybe we could stick all these arguments into a struct, and make that struct generic over the type of commitment? This will fix the clippy lint. Denote this struct `VoteMetaData<T>`. 

We could then type alias `pub type QcVote<WhateverSelfIs> = VoteMetaData<WhateverSelfIs::LeafType` and `pub type DACVote<WhateverSelfIs> = VoteMetaData<WhateverSelfIs::Block>`. Then this `accumulate_vote` could be made generic over the commitment type and certificate type, and we could have const function aliases as part of the trait that monomorphize over the struct type aliases and certificate types.

Reply:
Based of this suggestion I did something similar to this but wasn't sure about the type alias part.  LMK what you think

Reply:
The type aliases were meant as a convenience so we don't accidentally feed in the wrong commitment. But, it looks like Rust is inferring the type where needed, which is cleaner IMO. Nice, LGTM!